.github/guides/SMOOTHNESS.md

```diff
@@ -0,0 +1,75 @@
+
+This usually happens because the user's computer just doesn't have a powerful enough GPU (or CPU in some cases) to render a frame in the time required. This is something we can work on, but it's quite complicated since we lack effective profiling tools for clientside issues. You can get a decent feeling for your own machine by looking at the task manager (as a general rule: more stuff on screen/more filters being applied to that stuff -> more clientside lag. There's some flaws here on byond's end which lummox is working on but there's only so much you can do).
+
+This delay can also lead to a desync between client and server frames, which would cause the same sort of failure as server overrun.
+
+#### 3. Server Overrun
+
+The server is expected to finish a tick every tick_lag ds. If it CAN'T, like if the mc/a sleeping proc/a verb (more on these later) takes more then that amount of time to execute, then that'll show up as a delayed appearance update on the client, which means a jitter in any animations that are extended by the server (movement falls into this)
+
+If the overrun is consistent enough we can build this delay into our animations. This is most relevant for smooth movement (fancy byond stuff that functionally just animates a glide between two tiles). If we know the average overrun we can use it as a multiplier for how long animations like movement should take, and sync them back up with the server's functional tickrate. If the overrun isn't consistent then we just kind of have to bite it. This is one of the more persistant problems we have, since in theory any part of the codebase could overconsume the tick.
+
+#### 4. Server Underrun
+
+I've been lying to you a little bit so far. We don't actually need the server and client tick rate to sync up perfectly. What we actually need is the time between map sends to sync up with the client's framerate (IE, the time between mapsends needs to be consistent). This is because appearance info is sent by the server as soon as map sending finishes, the time after when verbs run is something closer to "dead" time (verbs don't even show up on byond's cpu tracking tools, although that's for different reasons).
+
+This means that if say, it takes 70% of the tick to get to sending maps on tick 1, 20% on tick 2, and 90% on tick 3, then clients will see that as jitter "forward" (50% of a tick separates 1 and 2) and then a long delay (170% of a tick separates 2 and 3).
+
+There are ways for byond to fix this jumping "foward" (it could just hold the frame for a little while, though this would be weird with afformentioned network problems), but fixing the delay is harder and would in theory require some form of clientside prediction.
```

Comment:
```suggestion
There are ways for byond to fix this jumping "forward" (it could just hold the frame for a little while, though this would be weird with aforementioned network problems), but fixing the delay is harder and would in theory require some form of client-side prediction.
```